### ISSUE TITLE
Unable to detect valid SELECT statement

### ISSUE THREAD

**Issue by michael-s-molina** [apache/superset#29448]
### Bug description

Our query parser is unable to detect the following valid SELECT statement:

```
WITH my_table AS (
  SELECT
    *
  FROM public.cleaned_sales_data
)
SELECT
  *
FROM my_table
```

On SQL Lab, it throws `Only SELECT statements are allowed against this database`.

### How to reproduce the bug

1 - Go to SQL Lab and try to execute the query
2 - Check the error message

### Screenshots/recordings

<img width="1778" alt="Screenshot 2024-07-01 at 17 15 36" src="https://github.com/apache/superset/assets/70410625/e6118ac8-5ba1-4da6-88e8-43926180cc60">

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Additional context

The error comes from: https://github.com/apache/superset/blob/028665030f7851f3db4e90ee5d483ddb09bf384c/superset/sql_parse.py#L926

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by Habeeb556** [comment]
#28145 is related I think, as applying the `sql_parse.patch` resolved this error for me.

**Comment by michael-s-molina** [comment]
It seems this was fixed by 4.1
